component/otelcol/auth/oauth2/oauth2_test.go

```diff
@@ -0,0 +1,155 @@
+package oauth2_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/component/otelcol/auth"
+	"github.com/grafana/agent/component/otelcol/auth/oauth2"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/grafana/agent/pkg/util"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config/configauth"
+	"gotest.tools/assert"
+)
+
+func TestConfigDefaults(t *testing.T) {
```

Comment:
This test currently fails with this error:
```
expected: oauth2.Arguments{ClientID:"someclientid", ClientSecret:"someclientsecret", TokenURL:"https://example.com/oauth2/default/v1/token", EndpointParams:url.Values{}, Scopes:[]string{}, TLSSetting:otelcol.TLSClientArguments{CAFile:"", CertFile:"", KeyFile:"", MinVersion:"", MaxVersion:"", ReloadInterval:0, Insecure:false, InsecureSkipVerify:false, ServerName:""}, Timeout:0}

actual  : oauth2.Arguments{ClientID:"someclientid", ClientSecret:"someclientsecret", TokenURL:"https://example.com/oauth2/default/v1/token", EndpointParams:url.Values(nil), Scopes:[]string(nil), TLSSetting:otelcol.TLSClientArguments{CAFile:"", CertFile:"", KeyFile:"", MinVersion:"", MaxVersion:"", ReloadInterval:0, Insecure:false, InsecureSkipVerify:false, ServerName:""}, Timeout:0}
```
In the documentation I'd set the default values for `endpoint_params` to `{}` and the value of `scopes ` to `[]`.

Do you think it's a problem that the defaults we mention in the documentation are not actually equivalent to omitting the value?

Reply:
I don't think it's an issue, there's not too many differences from the config perspective between nil and an empty slice/map. You could set your explicit default values to `null` when that should be the default, but I can't remember if we allocate things anyway.

Reply:
Ok, thank you, I'll just delete this unit test in this case.